platform/default/local_file_source.cpp

```diff
@@ -59,26 +60,17 @@ LocalFileSource::LocalFileSource()
 LocalFileSource::~LocalFileSource() = default;
 
 std::unique_ptr<AsyncRequest> LocalFileSource::request(const Resource& resource, Callback callback) {
-    class LocalFileRequest : public AsyncRequest {
-    public:
-        LocalFileRequest(Resource resource_, FileSource::Callback callback_, ActorRef<LocalFileSource::Impl> fs_)
-            : mailbox(std::make_shared<Mailbox>(*util::RunLoop::Get()))
-            , fs(fs_) {
-            fs.invoke(&LocalFileSource::Impl::request, resource_.url, [callback_, ref = ActorRef<LocalFileRequest>(*this, mailbox)](Response res) mutable {
-                ref.invoke(&LocalFileRequest::runCallback, callback_, res);
-            });
-        }
+    auto mailbox = std::make_shared<Mailbox>(*util::RunLoop::Get());
```

Comment:
I don't think it is possible. We share the `Scheduler`, but sharing a `Mailbox` would imply that the object would still be scheduled after its destruction. But thanks for pointing this out, because I just realized we are not closing the `Mailbox` on the `FileSourceRequest` destructor.